### ISSUE TITLE
E_WARNING - strpos(): Empty needle

### ISSUE THREAD

**Issue by awrog** [iusvar/grav-plugin-pdf-js#9]
Using grav-plugin-pdf-js-0.2.0.zip on Grav (Current v1.5.9) - Learn2 theme
pdf in data directory and stored with page

Whatever I try, I get no result or an error.

[pdfjs file = handleiding-LB-2.3.pdf]
PDFJs: Could not resolve file name 'handleiding-LB-2.3.pdf'.

[pdfjs=handleiding-LB-2.2.pdf]
PDFJs: Could not resolve file name 'handleiding-LB-2.2.pdf'.

[pdfjs file=data:handleiding-LB-2.3.pdf]
E_WARNING - strpos(): Empty needle
Log:
```
[2019-04-08 14:13:58] grav.CRITICAL: strpos(): Empty needle - Trace: #0 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Debugger.php(314): Whoops\Run->handleError(2, 'strpos(): Empty...', '/home/lrpdoc/do...', 54) #1 [internal function]: Grav\Common\Debugger->deprecatedErrorHandler(2, 'strpos(): Empty...', '/home/lrpdoc/do...', 54, Array) #2 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/pdf-js/PDFJsShortcode.php(54): strpos('/home/lrpdoc/do...', '') #3 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(137): Grav\Plugin\Shortcodes\PDFJsShortcode->process(Object(Thunder\Shortcode\Shortcode\ProcessedShortcode)) #4 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(105): Thunder\Shortcode\Processor\Processor->processHandler(Object(Thunder\Shortcode\Shortcode\ParsedShortcode), Object(Thunder\Shortcode\Processor\ProcessorContext), Array) #5 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(52): Thunder\Shortcode\Processor\Processor->processIteration('<p>[size=14]gee...', Object(Thunder\Shortcode\Processor\ProcessorContext), NULL) #6 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/classes/ShortcodeManager.php(227): Thunder\Shortcode\Processor\Processor->process('<p>[size=14]gee...') #7 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/shortcode-core.php(107): Grav\Plugin\ShortcodeManager->processContent(Object(Grav\Common\Page\Page), Object(Grav\Common\Data\Data)) #8 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Grav\Plugin\ShortcodeCorePlugin->onPageContentProcessed(Object(RocketTheme\Toolbox\Event\Event), 'onPageContentPr...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #9 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #10 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #11 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(279): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #12 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Page/Page.php(669): Grav\Common\Grav->fireEvent('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #13 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/shortcode-core.php(154): Grav\Common\Page\Page->content() #14 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Grav\Plugin\ShortcodeCorePlugin->onPageInitialized(Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #15 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #16 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #17 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(279): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #18 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Processors/PagesProcessor.php(26): Grav\Common\Grav->fireEvent('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #19 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(131): Grav\Common\Processors\PagesProcessor->process() #20 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(370): Grav\Common\Grav->Grav\Common\{closure}() #21 [internal function]: Grav\Common\Grav::Grav\Common\{closure}('pages', 'Pages', Object(Closure)) #22 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(346): call_user_func_array(Object(Closure), Array) #23 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(132): Grav\Common\Grav->__call('measureTime', Array) #24 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/index.php(54): Grav\Common\Grav->process() #25 {main} [] []
```

**Comment by iusvar** [comment]
If the file is in `user/data` you can use `[pdfjs file="data:handleiding-LB-2.3.pdf"]`

**Comment by awrog** [comment]
Thanks iusvar for reaching out to me.

I still get the same error:
```
[2019-04-09 07:27:07] grav.CRITICAL: strpos(): Empty needle - Trace: #0 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Debugger.php(314): Whoops\Run->handleError(2, 'strpos(): Empty...', '/home/lrpdoc/do...', 54) #1 [internal function]: Grav\Common\Debugger->deprecatedErrorHandler(2, 'strpos(): Empty...', '/home/lrpdoc/do...', 54, Array) #2 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/pdf-js/PDFJsShortcode.php(54): strpos('/home/lrpdoc/do...', '') #3 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(137): Grav\Plugin\Shortcodes\PDFJsShortcode->process(Object(Thunder\Shortcode\Shortcode\ProcessedShortcode)) #4 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(105): Thunder\Shortcode\Processor\Processor->processHandler(Object(Thunder\Shortcode\Shortcode\ParsedShortcode), Object(Thunder\Shortcode\Processor\ProcessorContext), Array) #5 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/vendor/thunderer/shortcode/src/Processor/Processor.php(52): Thunder\Shortcode\Processor\Processor->processIteration('<p>[size=14]gee...', Object(Thunder\Shortcode\Processor\ProcessorContext), NULL) #6 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/classes/ShortcodeManager.php(227): Thunder\Shortcode\Processor\Processor->process('<p>[size=14]gee...') #7 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/shortcode-core.php(107): Grav\Plugin\ShortcodeManager->processContent(Object(Grav\Common\Page\Page), Object(Grav\Common\Data\Data)) #8 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Grav\Plugin\ShortcodeCorePlugin->onPageContentProcessed(Object(RocketTheme\Toolbox\Event\Event), 'onPageContentPr...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #9 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #10 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #11 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(279): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #12 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Page/Page.php(669): Grav\Common\Grav->fireEvent('onPageContentPr...', Object(RocketTheme\Toolbox\Event\Event)) #13 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/user/plugins/shortcode-core/shortcode-core.php(144): Grav\Common\Page\Page->content() #14 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Grav\Plugin\ShortcodeCorePlugin->onPageInitialized(Object(RocketTheme\Toolbox\Event\Event), 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\EventDispatcher)) #15 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #16 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php(23): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #17 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(279): RocketTheme\Toolbox\Event\EventDispatcher->dispatch('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #18 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Processors/PagesProcessor.php(26): Grav\Common\Grav->fireEvent('onPageInitializ...', Object(RocketTheme\Toolbox\Event\Event)) #19 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(131): Grav\Common\Processors\PagesProcessor->process() #20 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(370): Grav\Common\Grav->Grav\Common\{closure}() #21 [internal function]: Grav\Common\Grav::Grav\Common\{closure}('pages', 'Pages', Object(Closure)) #22 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(346): call_user_func_array(Object(Closure), Array) #23 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/system/src/Grav/Common/Grav.php(132): Grav\Common\Grav->__call('measureTime', Array) #24 /home/lrpdoc/domains/lrp-doc.pkn.nl/public_html/index.php(54): Grav\Common\Grav->process() #25 {main} [] []
```

**Comment by iusvar** [comment]
there is a new release that solves the problem

availability on GPM will be shortly

**Comment by leycec** [comment]
Can this be safely closed now? The [0.3.0 release notes](https://github.com/iusvar/grav-plugin-pdf-js/releases/tag/0.3.0) suggest "Yuppers!", which is always nice. :+1:
